### PR TITLE
Update examples to use spread matching

### PR DESCRIPTION
**Pull request by gregorybchris**

### CODE REVIEW

File: examples/0_home/list-map.ss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-map (v -> v * 2) [1, 2, 3, 4] 4
-. map = f -> xs ->
-  | 0 -> []
-  | n -> (map f xs (n - 1)) +< (f (xs@(n - 1)))
\ No newline at end of file
+map (v -> v * 2) [1, 2, 3, 4]
```
Comment (tekknolagi) [inline examples/0_home/list-map.ss:1]: We should include this in STDLIB and maybe even test it

**Review by tekknolagi — APPROVED** [review]

File: examples/0_home/list-filter.ss [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
-. filter = f -> xs ->
-  | 0 -> []
-  | n -> get_rest (f xs@(n - 1))
-  . get_rest =
-    | 0 -> filter f xs (n - 1)
-    | 1 -> filter f xs (n - 1) +< xs@(n - 1)
\ No newline at end of file
+filter lt5 [2, 8, 4, 5, 1, 6, 3, 7]
+. lt5 = x -> x < 5
+. filter = f ->
```
Comment (tekknolagi) [inline examples/0_home/list-filter.ss:3]: filter too

**Comment by tekknolagi** [comment]
Excited for you to update with new filter

**Comment by gregorybchris** [comment]
Updated in #56

**Pull request closed without merging**